packages/process/src/node/process-manager.ts

```diff
@@ -22,17 +29,38 @@ export class ProcessManager {
         return id;
     }
 
-    get(id: number): Process | undefined {
-        return this.processes.get(id);
+    /**
+     * Removes the process from this process manager. Invoking this method, will make
+     * sure that the process is terminated before eliminating it from the manager's cache.
+     *
+     * @param process the process to unregister from this process manager.
+     */
+    unregister(process: Process): void {
+        if (!process.killed) {
+            process.kill();
```

Comment:
So if a process is not killed first it will not be unregistred ?
So you have to call it twice ?

I think the return line should be removed.

Reply:
Why? Can you explain? Thank you!

Reply:
Well if it's not killed killed is called and returns before calling unregister

Reply:
I think you are wrong: `process.kill()` will invoke `processManager.unregister(this)` after setting `killed` to `true`.

Reply:
Ho yes that is quite weird however imo... I'm commenting in another comment on what I think should be done.

Reply:
> Ho yes that is quite weird however imo...

I am open to any other approach. Since you came up with the initial implementation of the processes and the process manager, maybe you could advise how to unregister the process automatically from the manager on termination. Any ideas, and then I can update the PR with that? Thank you!

Reply:
Yes see: https://github.com/theia-ide/theia/pull/841#discussion_r152343062
I think this will work fine